client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx

```diff
@@ -0,0 +1,296 @@
+import React, { Fragment, PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import _some from 'lodash/some';
+
+import AuthInput from './AuthInput';
+import { changePassword } from 'modules/user/model/actions';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import {
+  makeAbortablePromise,
+  validatePasswordStructure
+} from 'common/utils/helpers';
+import PendingButton from 'common/components/PendingButton';
+import { IntlPropType } from 'common/constants/propTypes';
+import { ValidationException } from 'common/exceptions/ValidationException';
+
+class PasswordChangeForm extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newConfirmPasswordValue: undefined,
+      higherLevelErrors: {
+        newConfirmPasswordValueError: '',
+        newPasswordError: ''
+      },
+      isFormValid: false,
+      isNewPasswordValid: false,
+      isPasswordConfirmValid: false,
+      isPasswordValid: false,
+      newPassword: '',
+      password: '',
+      passwordChanged: false,
+      pending: false,
+      changePasswordErrorId: ''
+    };
+  }
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onPasswordChange = (password, isValid) => {
+    const { changePasswordErrorId } = this.state;
+    const error = isValid ? '' : changePasswordErrorId;
+
+    this.setState(
+      {
+        password,
+        isPasswordValid: isValid,
+        changePasswordErrorId: error
+      },
+      this.comparePasswords
+    );
+  };
+
+  onNewPasswordChange = (newPassword, isValid) => {
+    const {
+      higherLevelErrors,
+      higherLevelErrors: { newPasswordError }
+    } = this.state;
+    const error = isValid ? '' : newPasswordError;
+
+    this.setState(
+      {
+        higherLevelErrors: {
+          ...higherLevelErrors,
+          newPasswordError: error
+        },
+        newPassword,
+        isNewPasswordValid: isValid
+      },
+      this.comparePasswords
+    );
+  };
+
+  onNewPasswordConfirmChange = (newConfirmPasswordValue, isValid) =>
+    this.setState(
+      { newConfirmPasswordValue, isPasswordConfirmValid: isValid },
+      this.comparePasswords
+    );
+
+  passwordValidator = value =>
+    validatePasswordStructure(value) ? '' : 'user.auth.input.password.invalid';
```

Comment:
Please see my comment about common validators.

Reply:
added to JIRA under EOC_408 task